tests/test_pretest.py

```diff
@@ -0,0 +1,25 @@
+import pytest
+
+
+@pytest.mark.pretest
+def test_pretest(api, utils):
+    """
+    This test does 2 things
+    1) Restore ports to factory defaults and keep them ready for tests
+    2) Remove any previous stale sessions in API server
+
+    """
+    config = api.config()
+
+    for i in range(len(utils.settings.ports)):
+        config.ports.port(name="port%d" % i, location=utils.settings.ports[i])[
+            -1
+        ]
+
```

Comment:
Have we tested that this works if we cancel pipeline while the test is running and then restart pipeline ?